### PR TITLE
Upgrade to go-ipfs-blockstore `v1.2.0`

### PR DESCRIPTION
**Pull request by masih**
Upgrade to go-ipfs-blockstore `v1.2.0` and reflect on breaking changes.

### CODE REVIEW

**Comment by masih** [comment]
Cc @jennijuju

File: cgo/blockstore.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 
 	blocks "github.com/ipfs/go-block-format"
 	"github.com/ipfs/go-cid"
-	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/ipfs/go-ipld-format"
```
Comment (arajasek) [inline cgo/blockstore.go:8]: Can we import this as `ipld`? I think it looks clearer that way (see, eg. [here](https://github.com/ipfs/go-ipfs-blockstore/pull/48/files#diff-d0909a447e05cb276100d4f9dd10ffcab536acd3d91d84111361df2188c4e186R163))

**Review by arajasek — APPROVED** [review]
SGWM, with one nit!

**Review by Stebalien — APPROVED** [review]

File: cgo/blockstore.go (hunk 1 above)
Comment (jennijuju) [reply cgo/blockstore.go:8]: @masih once addressed plz feel free to land!

**Comment by masih** [comment]
Thank you all for the reviews. I don't have permissions to merge this I'm afraid. 

Once merged I'd also need your help in tagging the commit please.

**Pull request merged**